spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/post/SendErrorFilter.java

```diff
@@ -76,6 +77,8 @@ public Object run() {
 				request.setAttribute("javax.servlet.error.message", message);
 			}
 
+			//unwrap request if needed, otherwise tomcat hangs in endless loop
```

Comment:
And there is the problem. getRequest() doesn't return the request that is wrapped by the current wrapper (i.e. req) it returns a new, wrapped version of req. That is a bug in zuul.